code/model/DNEnvironment.php

```diff
@@ -615,9 +615,18 @@ public function CurrentBuild() {
 
 	/**
 	 * A list of past deployments.
-	 * @return ArrayList
+	 * @param string $orderBy - the name of a DB column to sort in descending order
+	 * @return \ArrayList
 	 */
-	public function DeployHistory() {
+	public function DeployHistory($orderBy = '') {
+
+		$sort = [];
+		if($orderBy != '') {
+			$sort[$orderBy] = 'DESC';
+		}
+		// default / fallback sort order
+		$sort['LastEdited'] = 'DESC';
```

Comment:
This is not effective, you are not passing `$sha` into the ORM call underneath :)